network/p2p/libp2pConnector.go

```diff
@@ -129,6 +129,10 @@ func (l *libp2pConnector) trimAllConnectionsExcept(peerInfos []peer.AddrInfo) {
 
 		peerInfo := l.host.Network().Peerstore().PeerInfo(peerID)
 		log := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
+		if l.host.ConnManager().IsProtected(peerID, "") {
```

Comment:
if there is an attempt in progress to create a one-to-one stream then do not prune the connection